src/containers/Tenant/Diagnostics/TopShards/Filters/Filters.tsx

```diff
@@ -0,0 +1,56 @@
+import {RadioButton} from '@gravity-ui/uikit';
+
+import {DateRange, DateRangeValues} from '../../../../../components/DateRange';
+
+import {
+    EShardsWorkloadMode,
+    IShardsWorkloadFilters,
+} from '../../../../../types/store/shardsWorkload';
+
+import {isEnum} from '../../../../../utils/typecheckers';
+
+import i18n from '../i18n';
+import {b} from '../TopShards';
+
+import './Filters.scss';
+
+interface FiltersProps {
+    value: IShardsWorkloadFilters;
+    onChange: (value: Partial<IShardsWorkloadFilters>) => void;
+    className?: string;
+}
+
+export const Filters = ({value, onChange, className}: FiltersProps) => {
+    const handleModeChange = (mode: string) => {
+        if (!isEnum(EShardsWorkloadMode, mode)) {
+            const values = Object.values(EShardsWorkloadMode).join(', ');
+            throw new Error(`Unexpected TopShards mode "${mode}". Should be one of: ${values}`);
+        }
+
+        onChange({mode});
+    };
+
+    const handleDateRangeChange = (dateRange: DateRangeValues) => {
+        onChange({
+            mode: EShardsWorkloadMode.History,
+            ...dateRange,
+        });
+    };
+
+    const from = value.mode === EShardsWorkloadMode.Immediate ? undefined : value.from;
+    const to = value.mode === EShardsWorkloadMode.Immediate ? undefined : value.to;
+
+    return (
+        <div className={b('filters', className)}>
```

Comment:
Will it be better not to import `b` here, but pass className `b('filters')` via props as your styles are defined on TopShards level?

Reply:
no, it's this element name, and its styles are in Filters.scss